src/cev/_embedding_comparison_widget.py

```diff
@@ -46,11 +49,17 @@ def _create_titles(
     return spacer, title_widget
 
 
+def _coerce_to_embedding(embedding: Embedding | pd.DataFrame) -> Embedding:
+    if isinstance(embedding, Embedding):
+        return embedding
+    return Embedding.from_df(embedding)
+
+
 class EmbeddingComparisonWidget(ipywidgets.VBox):
     def __init__(
         self,
-        left_embedding: Embedding,
-        right_embedding: Embedding,
+        left_embedding: Embedding | pd.DataFrame,
+        right_embedding: Embedding | pd.DataFrame,
```

Comment:
Is this backwards compatible with Python versions before `11`?

Reply:
Yes, the top of the file uses:


```python
from __future__ import annotations
```

Which tells Python to treat everything after the `:` in types as comments. We only use for static type checking, so this allows us to use modern syntax. If we needed to inspect the types at runtime, we'd want to avoid using `__future__` annotations imports.